javav2/usecases/creating_fsa_app/src/main/java/com/example/fsa/services/ExtractTextService.java

```diff
@@ -13,41 +14,57 @@
 import software.amazon.awssdk.services.textract.model.Block;
 import software.amazon.awssdk.services.textract.TextractClient;
 import software.amazon.awssdk.services.textract.model.S3Object;
+import software.amazon.awssdk.services.textract.model.TextractException;
 
 public class ExtractTextService {
 
+    private static TextractClient textractClient;
+
+    private static synchronized TextractClient getTextractClient() {
```

Comment:
If you used a static initializer, this would happen during the function's init phase. (It's probably trivially cheap, but IMHO we should be showing good practice for Lambdas, as they're hard enough already. If this were a larger task, like initializing a db conn pool, or verifying precondition state, keeping it during function init rather will keep it accounted better, rather than marking it as happening during the first invocation.)

Reply:
This has all been modified